test/dub_package/source/app.d

```diff
@@ -43,3 +43,119 @@ unittest
     scope parser = new Parser!ASTBase(null, null, false);
     assert(parser !is null);
 }
+
+
+// inits the DMD compiler
+void init()
+{
+    import ddmd.globals : global;
+
+    global._init;
+    global.params.isLinux = 1;
+    import ddmd.dmodule : Module;
+    import ddmd.id : Id;
+    import ddmd.mtype : Type;
+    import ddmd.target : Target;
+    import ddmd.expression : Expression;
+    import ddmd.objc : Objc;
+    import ddmd.builtin : builtin_init;
+
+    // Initialization
+    Type._init();
+    Id.initialize();
+    Module._init();
+    Target._init();
+    Expression._init();
+    Objc._init();
+    builtin_init();
+}
+
+// add import paths
+void addImports(string[] path)
+{
+    import ddmd.globals : global;
+    import ddmd.arraytypes : Strings;
+
+    Strings* res = new Strings();
+    foreach (p; path)
+    {
+        import std.string : toStringz;
+        Strings* a = new Strings();
+        a.push(p.toStringz);
+        res.append(a);
+    }
+    global.path = res;
+}
```

Comment:
Well, obviously it would be nice if the `dmd` "library` contains a couple of high-level wrapper, s.t. a potential author only has to interact with its shiny, modern D APIs.